torch/csrc/jit/cuda/cuda.h

```diff
@@ -0,0 +1,177 @@
+#include <aten/src/ATen/cuda/CUDAEvent.h>
+#include <c10/core/Device.h>
+#include <c10/cuda/CUDAStream.h>
+#include <torch/custom_class.h>
+
+namespace torch {
+namespace jit {
+
+class CUDAEvent;
+// This class is a wrapper around c10::cuda::CUDAStream.
+// It is needed because TorchBind does not support all of the argument types
+// for c10::cuda::CUDAStream. For more details, please refer to c10/cuda/CUDAStream.h.
+class CUDAStream final : public CustomClassHolder {
```

Comment:
Couldn't the same be achieved by multiple inheritance?

Reply:
Making `c10::cuda::CUDAStream` inherit from `CustomClassHolder` would not be enough; we would also have to modify that class so that all of the args are types that fit into an `IValue`. For things like device index that's already true, but any arguments of type `CUDAStream` or `CUDAStream&` have to be converted to `c10::intrusive_ptr<CUDAEvent>` instead so that TorchBind is happy.